### ISSUE TITLE
App factory project structure with GunicornInternalPrometheusMetrics()

### ISSUE THREAD

**Issue by mamor1** [rycus86/prometheus_flask_exporter#33]
I have an app that I want to run replicas of in a K8s cluster. For this I am using the GunicornInternalPrometheusMetrics class like so inside `__init__.py`:

```
from flask import Flask
from prometheus_flask_exporter.multiprocess import GunicornInternalPrometheusMetrics

def create_app():
     app = Flask(__name__)
     metrics = GunicornInternalPrometheusMetrics(app)
```
In the app file I define the app object like so:
```
from app import create_app

app = create_app(api_config or "default")
```

However, this does not make the metrics decorator available for my modules (which have the routes).

Therefore, I tried to make it in the app factory way:
```
from flask import Flask
from prometheus_flask_exporter.multiprocess import GunicornInternalPrometheusMetrics

mnetrics = GunicornInternalPrometheusMetrics(app=None)

def create_app():
     app = Flask(__name__)
     metrics.init_app(app)
```

However, this causes flask to complain that I am working outside of the context.

What is the preferred way to do this so that the decorator (ie. something like `@metrics.gauge('in_progress', 'Long running requests in progress')`) is possible?

**Comment by rycus86** [comment]
Hi @mamor1 !
Thanks a lot for bringing this to my attention! There was a bug, with `app=None` in this type of metrics implementation, it should be fixed now.
You can see the example here (https://github.com/rycus86/prometheus_flask_exporter/blob/master/examples/gunicorn-app-factory/app_setup.py), I basically did what you tried above. With the bugfix it will now work.
Once the change goes through Travis, it will be available in version `0.9.1` (in about 15 minutes).

Thanks again!

**Comment by kotarf** [comment]
@rycus86 hello..I'm having issues with applying the app factory pattern to both GunicornInternalPrometheusMetrics and GunicornPrometheusMetrics in 0.12.1. I'm not getting any metrics recorded unless I directly instantiate with the _app_ argument provided (so no factory pattern).

 Is there any way you could describe the bug fix so I can attempt to troubleshoot locally? Thank you.
